js/ui/grid_core/ui.grid_core.state_storing.js

```diff
@@ -172,8 +172,8 @@ export const stateStoringModule = {
                         exportController.selectionOnly(state.exportSelectionOnly);
                     }
 
-                    if(selectedRowKeys) {
-                        that.option('selectedRowKeys', selectedRowKeys);
+                    if(!that.option('selection.deferred')) {
```

Comment:
In case `selection.deferred` is `true` and `applyState` is called when page is just reloaded, calling `that.option('selectedRowKeys', ...)` leads to error [E1042](https://js.devexpress.com/Documentation/ApiReference/UI_Components/Errors_and_Warnings/#E1042) "`keyExpr` is not set". 

It happens because state is applied before dataSource is initialized (See [`_refreshDataSource` overloading in state storing](_refreshDataSource), but changing `selectedRowKeys` in deferred mode requires `keyExpr`

It seems that `selectionFilter` replicates current selected nodes, so I added `!that.option('selection.deferred')` condition